dd-trace-core/src/main/java/datadog/trace/core/PendingTraceBuffer.java

```diff
@@ -277,4 +316,52 @@ public static PendingTraceBuffer discarding() {
   public abstract void flush();
 
   public abstract void enqueue(Element pendingTrace);
+
+  private static class TracerDump implements TracerFlare.Reporter {
+    private static final Comparator<Element> TRACE_BY_START_TIME =
+        comparingLong(trace -> trace.getRootSpan().getStartTime());
+    private static final Predicate<Element> NOT_PENDING_TRACE =
+        element -> !(element instanceof PendingTrace);
+    private final DelayingPendingTraceBuffer buffer;
+
+    private TracerDump(DelayingPendingTraceBuffer buffer) {
+      this.buffer = buffer;
+    }
+
+    @Override
+    public void prepareForFlare() {
+      if (buffer.worker.isAlive()) {
+        int count = buffer.dumpCounter.get();
+        int loop = 1;
+        boolean signaled = buffer.queue.offer(DelayingPendingTraceBuffer.DUMP_ELEMENT);
+        while (!buffer.closed && !signaled) {
+          buffer.yieldOrSleep(loop++);
+          signaled = buffer.queue.offer(DelayingPendingTraceBuffer.DUMP_ELEMENT);
+        }
+        int newCount = buffer.dumpCounter.get();
+        while (!buffer.closed && count >= newCount) {
+          buffer.yieldOrSleep(loop++);
+          newCount = buffer.dumpCounter.get();
+        }
+      }
+    }
+
+    @Override
+    public void addReportToFlare(ZipOutputStream zip) throws IOException {
+      DelayingPendingTraceBuffer.DumpDrain.DATA.removeIf(NOT_PENDING_TRACE);
+      // Storing oldest traces first
+      DelayingPendingTraceBuffer.DumpDrain.DATA.sort((TRACE_BY_START_TIME).reversed());
```

Comment:
```suggestion
      DelayingPendingTraceBuffer.DumpDrain.DATA.sort(TRACE_BY_START_TIME.reversed());
```
You could also move the `.reversed()` call to where you build the comparator and rename the constant to be `OLDEST_TRACE_FIRST`